specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,429 @@
+/// only allows to select `DEFAULT`; otherwise, will deny the download
+/// and return `COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_NOT_SUPPORTED`.
+///
+/// The content type/format is a MIME type, indicated by the source
+/// server side and identified by the browser. It’s not related to the
+/// file’s type or extension. MIME type of `text/html`,
+/// `application/xhtml+xml` are considered as html page.
+[v1_enum] typedef enum COREWEBVIEW2_SAVE_AS_REQUESTED_KIND {
+  /// Default to save for a non-html content. If it is selected for a html
+  /// page, it’s same as HTML_ONLY option.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_DEFAULT,
+  /// Save the page as html
+  COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_HTML_ONLY,
+  /// Save the page as mhtml
+  COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_SINGLE_FILE,
+  /// Save the page as html, plus, download the page related source files in
+  /// a same name directory
+  COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_COMPLETE,
+} COREWEBVIEW2_SAVE_AS_REQUESTED_KIND;
+
+/// Status of a programmatic save as call, indicates the result
+/// for method `RequestSaveAs`
+[v1_enum] typedef enum COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS {
+  /// Could not perform Save As because the destination file path is an invalid path.
+  ///
+  /// It is considered as invalid when:
+  /// the path is empty, a relativate path, the parent directory doesn't
+  /// exist, or the path is a driectory.
+  ///
+  /// Parent directory can be itself, if the path is root directory, or
+  /// root disk. When the root doesn't exist, the path is invalid.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_INVALID_PATH,
+  /// Could not perform Save As because the destination file path already exists and 
+  /// replacing files was not allowed by the `AllowReplace` property.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_FILE_ALREADY_EXISTS,
+  /// Could not perform Save As when the `Kind` property selection not
+  /// supported because of the content MIME type or system limits
+  ///
+  /// MIME type limits please see the emun `COREWEBVIEW2_SAVE_AS_REQUESTED_KIND`
+  ///
+  /// System limits might happen when select `HTML_ONLY` for an error page,
+  /// select `COMPLETE` and WebView running in an App Container, etc.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_NOT_SUPPORTED,
+  /// Did not perform Save As because the client side decided to cancel.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_CANCELLED,
+  /// Save as requested completeed, the downloading would start
+  COREWEBVIEW2_SAVE_AS_REQUESTED_COMPLETED,
+} COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS;
+
+[uuid(15e1c6a3-c72a-4df3-91d7-d097fbec3bfd), object, pointer_default(unique)]
+interface ICoreWebView2_20 : IUnknown {
+  /// Programmatically trigger a save as action for current content. `SaveAsRequested` 
+  /// event will be raised.
+  ///
+  /// Opens a system modal dialog by default. If it was already opened, this method 
+  /// would not open another one. If the `SuppressDefaultDialog` is TRUE, won't open 
```

Comment:
'this method would not open another one' will leave folks wondering what happens to the SaveContentAs method. Does it return a failed HRESULT or a different result value to indicate this issue?